### ISSUE TITLE
DOS"$//dirname"

### ISSUE THREAD

**Issue by bobbyjim** [commanderx16/x16-rom#170]
Here's another mostly-frivolous command extension: list a subdirectory's contents without having to first change into that subdirectory.  This is somewhat symmetric with the ability to SCRATCH a file in a subdirectory without having to CD there first.

I haven't seen a way to do this, hence this suggestion.

**Comment by BruceMcF** [comment]
In CMD syntax, // is the root directory.

More in line with CMD syntax would be: "$/dirname/", if it is a subdirectory of the current directory.

To get an idea of existing practice, [CMD's FD4000 user manual (pdf)](http://www.zimmers.net/anonftp/pub/cbm/manuals/cmd/CMD_FD2000_Manual.zip) is at:
http://www.zimmers.net/anonftp/pub/cbm/manuals/cmd/index.html

**Comment by bobbyjim** [comment]
Either or both would be nice.
DOS"$/dirname/"  to list the subdirectory of the current directory.
DOS"$//dirname/" to list the subdirectory of root.

UNIX has both.  I can say
ls foo
ls /foo

**Comment by BruceMcF** [comment]
Yes, both would be the syntax used in CMD drives ... /dirname1/dirname2/ is the subdirectory dirname1 of the current directory and ITS subdirectory dirname2 ... //dirname1/dirname2/ is the subdirectory dirname1 in the root directory, and its subdirectory dirname2.

The jiffydos ^ (uparrow) is shorthand for LOAD"//: so ^* loads the first file in the root directory. It might run it as well - I had WarpSpeed not JiffyDOS, so I am a little hazy on some of the CMD extensions to classical Wedge syntax.

**Comment by greg-king5** [comment]
JiffyDOS doesn't change any of the commands that are in Commodore's DOS Wedge.  `^` loads and runs a BASIC program.  It doesn't preset the pathname.  Therefore, you would see:
```
^
?missing file name  error
```

**Comment by BruceMcF** [comment]
Thanks for that, though you seem to have skated past the "the syntax used in CMD drives" part of the comment ... that's not due to the wedge, that's due to the device on the other side of the IEC bus.

... if ^ loads and runs the file that is found, then it seems like ^//* really would load and run the first file in the root directory, since CMD drives seem to (helpfully) shortcut the "//:filename" syntax for "//*".
